### PR TITLE
admission: do not leak admission plugin config types outside of the plugin

### PR DESCRIPTION
**Pull request by sttts**
We had two plugins that did not register their types into the external config scheme, breaking embedded config for them without noticing. Actually though, we do not want to leak those types at all into external scheme, but keep the types internal to the plugins. The second commit in this PR does that.

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/admission/config.go [hunk 1]
```diff
@@ -131,12 +131,17 @@ func GetAdmissionPluginConfigurationFor(pluginCfg apiserver.AdmissionPluginConfi
 	// if there is nothing nested in the object, we return the named location
 	obj := pluginCfg.Configuration
 	if obj != nil {
-		// serialize the configuration and build a reader for it
-		content, err := writeYAML(obj, scheme)
-		if err != nil {
-			return nil, err
+		switch obj := obj.(type) {
+		case *runtime.Unknown:
```
Comment (mfojtik) [inline staging/src/k8s.io/apiserver/pkg/admission/config.go]: maybe deserves a comment

**Comment by mfojtik** [comment]
/lgtm

cc @deads2k

File: staging/src/k8s.io/apiserver/pkg/admission/config.go (hunk 1 above)
Comment (sttts) [reply staging/src/k8s.io/apiserver/pkg/admission/config.go]: done

**Comment by deads2k** [comment]
The approach seems reasonable, but we have more registered now.  Hiding the scheme ought to flush them all out.

**Pull request merged**